pkg/tasks/queue/pgqueue.go

```diff
@@ -669,6 +669,9 @@ func (p *PgQueue) Requeue(taskId uuid.UUID) error {
 	if err == pgx.ErrNoRows {
 		return ErrNotExist
 	}
+	if info.CancelAttempted {
+		return ErrTaskCanceled
+	}
```

Comment:
One more thing to add here, I would also check that the task isn't "running" before returning this error.

This Requeue method is currently used by the tasking system to requeue running tasks. We wouldn't want to prevent a "running" task from re-queuing here if it were canceled, because then it would be stuck in the "running" state.

Reply:
This isn't a realistic scenario, but it could be an edge case